### DISCUSSION TITLE
cannot detect text " Since the angle classifier is not initialized, it will not be used during the forward process "

### DISCUSSION THREAD

**Discussion by namaws** [PaddlePaddle/PaddleOCR#12960] [Q&A]
Hello,

I want to use paddleocr to detect text from image. However I keep getting this warning and not really detecting any text. 
(the testing image here was from an example of people using paddleocr to do text recognition, so it should be able to detect)

```ocr_model = PaddleOCR(lang='en', cls=False)```   

ppocr DEBUG: Namespace(help='==SUPPRESS==', use_gpu=False, use_xpu=False, use_npu=False, ir_optim=True, use_tensorrt=False, min_subgraph_size=15, precision='fp32', gpu_mem=500, gpu_id=0, image_dir=None, page_num=0, det_algorithm='DB', det_model_dir='C:\\Users\\TSECIN94/.paddleocr/whl\\det\\en\\en_PP-OCRv3_det_infer', det_limit_side_len=960, det_limit_type='max', det_box_type='quad', det_db_thresh=0.3, det_db_box_thresh=0.6, det_db_unclip_ratio=1.5, max_batch_size=10, use_dilation=False, det_db_score_mode='fast', det_east_score_thresh=0.8, det_east_cover_thresh=0.1, det_east_nms_thresh=0.2, det_sast_score_thresh=0.5, det_sast_nms_thresh=0.2, det_pse_thresh=0, det_pse_box_thresh=0.85, det_pse_min_area=16, det_pse_scale=1, scales=[8, 16, 32], alpha=1.0, beta=1.0, fourier_degree=5, rec_algorithm='SVTR_LCNet', rec_model_dir='C:\\Users\\TSECIN94/.paddleocr/whl\\rec\\en\\en_PP-OCRv4_rec_infer', rec_image_inverse=True, rec_image_shape='3, 48, 320', rec_batch_num=6, max_text_length=25, rec_char_dict_path='C:\\Users\\TSECIN94\\AppData\\Local\\anaconda3\\envs\\paddle-env\\Lib\\site-packages\\paddleocr\\ppocr\\utils\\en_dict.txt', use_space_char=True, vis_font_path='./doc/fonts/simfang.ttf', drop_score=0.5, e2e_algorithm='PGNet', e2e_model_dir=None, e2e_limit_side_len=768, e2e_limit_type='max', e2e_pgnet_score_thresh=0.5, e2e_char_dict_path='./ppocr/utils/ic15_dict.txt', e2e_pgnet_valid_set='totaltext', e2e_pgnet_mode='fast', use_angle_cls=False, cls_model_dir='C:\\Users\\TSECIN94/.paddleocr/whl\\cls\\ch_ppocr_mobile_v2.0_cls_infer', cls_image_shape='3, 48, 192', label_list=['0', '180'], cls_batch_num=6, cls_thresh=0.9, enable_mkldnn=False, cpu_threads=10, use_pdserving=False, warmup=False, sr_model_dir=None, sr_image_shape='3, 32, 128', sr_batch_num=1, draw_img_save_dir='./inference_results', save_crop_res=False, crop_res_save_dir='./output', use_mp=False, total_process_num=1, process_id=0, benchmark=False, save_log_path='./log_output/', show_log=True, use_onnx=False, output='./output', table_max_len=488, table_algorithm='TableAttn', table_model_dir=None, merge_no_span_structure=True, table_char_dict_path=None, layout_model_dir=None, layout_dict_path=None, layout_score_threshold=0.5, layout_nms_threshold=0.5, kie_algorithm='LayoutXLM', ser_model_dir=None, re_model_dir=None, use_visual_backbone=True, ser_dict_path='../train_data/XFUND/class_list_xfun.txt', ocr_order_method=None, mode='structure', image_orientation=False, layout=True, table=True, ocr=True, recovery=False, use_pdf2docx_api=False, invert=False, binarize=False, alphacolor=(255, 255, 255), lang='en', det=True, rec=True, type='ocr', ocr_version='PP-OCRv4', structure_version='PP-StructureV2', cls=False) 

```result = ocr_model.ocr(img_path)```    
 
ppocr WARNING: Since the angle classifier is not initialized, it will not be used during the forward process
[2024/03/25 09:28:53] ppocr DEBUG: dt_boxes num : 6, elapsed : 0.1299896240234375
[2024/03/25 09:28:53] ppocr DEBUG: rec_res num  : 6, elapsed : 0.6188290119171143

- 系统环境/System Environment：Window 10 
- 版本号/Version：Paddle：2.6.0  PaddleOCR： 2.7.2  Python: 3.11.8
- (using CPU)

**Comment by tink2123** [comment]
`ppocr WARNING: Since the angle classifier is not initialized, it will not be used during the forward process`  is just a warning. 
Looking at the output log, 6 text boxes have been successfully predicted. You can continue to print the `result` to confirm the correctness.

**Comment by gun-hash** [comment]
hey, i was facing the same thing. you said it was just a warning. can you tell me more about what the difference in initializing or not initializing this angle classifier?

**Comment by CosmicScot** [comment]
I got rid of Debug and ppocr WARNING messages using the below code:

# Initialize PaddleOCR with English language support
ocr = PaddleOCR(use_angle_cls=True, lang='en', show_log = False)
